csrc/scheduler/reduction_heuristic.h

```diff
@@ -133,8 +135,22 @@ class ReductionParams : public HeuristicParams {
   // block_dim_inner_reduction_extra (usually TIDy)
   ParallelType block_dim_inner_reduction_extra = ParallelType::Serial;
 
-  // use shared memory for persistent buffer, if false, will use registers
+  // If false, only register is used to store persistent buffers.
```

Comment:
Why do we need two properties? Why not just `smem_persistent_tvs` and let it mean tensors in the vector are stored in shared memory?

Reply:
> Why do we need two properties?

It's not strictly required. They are there because the current **inner persistent scheduler must move all or none** of the persistent buffers to smem. So there was only one `bool` indicating moving all or nothing, which is more efficient than using a vector of tensors. However, for this innerOuter scheudler, we needs to identify which one is being moved, so `smem_persistent_tvs` is added.

> Why not just smem_persistent_tvs and let it mean tensors in the vector are stored in shared memory?

we can do that. It means we need to change the inner persistent scheduler. I was trying to make this PR focus on the innerOuter persistent scheduler and prefer to leave the changes to other schedulers to other PRs if possible.

Of course, I can also add the changes into this PR if you feel that is more appropriate. :)

Added a TODO comment:
```
  // TODO: For innerPersistentHeuristic, all or none of the persistent buffers
  // should be stored in shared memory. The code can be extended to allow moving
  // some of the persistent buffers to shared memory while others still in
  // registers, similar to what we did in the innerOuterPersistentHeuristic.
```

Reply:
> So there was only one bool indicating moving all or nothing, which is more efficient than using a vector of tensors. 
I doubt the efficiency difference is significant, especially since the vector length, i.e., the number of persistent buffers, should be quite small. It'd be more important to make the heuristics as much simple as possible. It should be more preferable to extend an existing parameter rather than adding a new parameter.

Reply:
If I read Liqiang's comment correctly, I think the bigger impact from removing `shared_mem_persistent_buffer` is that vvv
> It means we need to change the inner persistent scheduler.

though I"m not sure how many LOC that means... If it's small let's do it here. Otherwise we can open a separate PR for clean up.

Reply:
Shouldn't it be just replacing `shared_mem_persistent_buffer` with `!smem_persistent_tvs.empty()`?

Reply:
Yeah for its uses in this PR it's straightforward.

For inner persistent scheduler, we need to throw something in `smem_persistent_tvs`, and I'm assuming we'd want to put something reasonable there as well. That's where I'm not sure how much change is needed. Maybe it's just a couple lines and in that case we should squeeze it in this PR as well.

Reply:
I added a commit deleting `shared_mem_persistent_buffer`. The section of code to move buffer to smem is put into a utility function shared by both inner persistent and innerOuter persistent.